src/parachain/nomination.ts

```diff
@@ -74,54 +77,53 @@ export interface NominationAPI extends TransactionAPI {
     isVaultOptedIn(accountId: string): Promise<boolean>;
     /**
      * @remarks At least one of the parameters must be specified
-     * @param currency The currency of the nominations
      * @param nominatorId Id of user who nominated to one or more vaults
      * @param vaultId Id of vault who is opted in to nomination
-     * @returns A list of `[[nominatorId, vaultId], nominatedAmount]` tuples
+     * @returns A list of `[[nominatorId, vaultId], nominatedAmount, wrappedCurrency]` tuples
      */
-    getFilteredNominations<C extends CollateralUnit>(
-        currency: Currency<C>,
+    getFilteredNominations(
         nominatorId?: string,
         vaultId?: string
-    ): Promise<[[string, string], MonetaryAmount<Currency<C>, C>][]>;
+    ): Promise<[[string, string], MonetaryAmount<Currency<CollateralUnit>, CollateralUnit>, WrappedCurrency][]>;
     /**
      * @remarks At least one of the parameters must be specified
-     * @param currency The collateral currency of the nominations
+     * @param wrappedCurrency The currency towards whose issuance the nomination was made
```

Comment:
I am wondering if it would make more sense to add the wrapped currency as a default on instantiation of the lib. Since you'd use the lib with exactly one wrapped token (i.e., either kBTC or interBTC), I think it would make sense. Also in the case where one chain would serve different wrapped assets like BCH, ZEC, I think unlikely that you were to change the wrapped asset individually per call, but rather have separate instances of the lib for each wrapped token.

Reply:
I would also prefer this, there are far to many parameters anyway on some functions. For example, I found it difficult to workout what arguments were required for `issueSingle` when I was working on the tests. It would be nicer to adopt the builder pattern for these utils I think.

Reply:
The builder pattern sounds good, in a different PR though